src/luascript.cpp

```diff
@@ -8740,43 +8740,67 @@ int LuaScriptInterface::luaPlayerSendOutfitWindow(lua_State* L)
 	return 1;
 }
 
-int LuaScriptInterface::luaPlayerAddMount(lua_State* L)
-{
-	// player:addMount(mountId)
+int LuaScriptInterface::luaPlayerAddMount(lua_State* L) {
+	// player:addMount(mountId or mountName)
 	Player* player = getUserdata<Player>(L, 1);
-	if (player) {
-		uint8_t mountId = getNumber<uint8_t>(L, 2);
-		pushBoolean(L, player->tameMount(mountId));
-	} else {
+	if (!player) {
 		lua_pushnil(L);
+		return 1;
 	}
+
+	uint8_t mountId;
+	if (isString(L, 2)) {
```

Comment:
Yes, he should test for number instead, and just assume it's a string otherwise.